### PR TITLE
Various updates to top-level README

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/2850

 # Rationale for this change

The project has matured a lot and we need to promote all the great work and encourage more people to try DataFusion out

# What changes are included in this PR?

- More information on features and use cases
- Update list of projects using DataFusion
- Removed links to community object store extensions because those are now incompatible after the recent object store changes if I am understanding this correctly

# Are there any user-facing changes?

No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,52 +21,70 @@
...
+- [Ballista](https://github.com/apache/arrow-ballista) Distributed SQL Query Engine
 - [Blaze](https://github.com/blaze-init/blaze) Spark accelerator with DataFusion at its core
 - [Cloudfuse Buzz](https://github.com/cloudfuse-io/buzz-rust)
 - [Cube Store](https://github.com/cube-js/cube.js/tree/master/rust)
-- [delta-rs](https://github.com/delta-io/delta-rs)
+- [datafusion-tui](https://github.com/datafusion-contrib/datafusion-tui) Text UI for DataFusion
+- [delta-rs](https://github.com/delta-io/delta-rs) Native Rust implementation of Delta Lake
 - [Flock](https://github.com/flock-lab/flock)
 - [InfluxDB IOx](https://github.com/influxdata/influxdb_iox) Time Series Database
+- [qv](https://github.com/timvw/qv) Quickly view your data
```
Comment (andygrove) [inline README.md:87]: @timvw I assume you are ok with having this listed here?

**Review by xudong963 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -21,52 +21,70 @@
...
-and Parquet) using threads.
+## Features
 
-DataFusion also supports distributed query execution via the
-[Ballista](https://github.com/apache/arrow-ballista/) crate.
+- SQL query planner with support for multiple SQL dialects
+- DataFrame API
+- Parquet, CSV, JSON, and Avro file formats are supported natively. Custom
+  file formats can be supported by implementing a `TableProvider` trait.
+- Supports popular object stores, including AWS S3, Azure Blob
```
Comment (andygrove) [inline README.md:34]: @tustvold Have I stated this correctly? I am not familiar with the recent changes.

Comment (tustvold) [reply README.md:34]: Yes, it also supports GCS

**Review by tustvold — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (timvw) [reply README.md:87]: Yes. (Sorry for the late replies, enjoying holidays this month ;)

**Pull request merged**